### PR TITLE
V0.5.0.0

### LINKED ISSUE #253: [ldd-vgr] <earth_received_time needs to be nillable>
**Issue Type**
Using the LABELS section to the right of this window >>>> , please indicate if this is a BUG or ENHANCEMENT request. If unknown, feel free to leave that blank and the LDD Steward can triage appropriately.

**Describe the issue identified (if applicable)**
Within the VGR mission LDD, earth _recieve_time is a required attribute, but is not nillable. The source VGR data being used in our study does not always contain an earth _recieve_time.  WIthout input data we can not give a reliable time for this attribute.  WE therefore need earth _recieve_time to be nillable.

**Describe the solution you'd like**
Since there are instances in the source data where earth_recieve_time is not given, there is no way we can populate this attribute with a meaningful time.  It would be best to make this attribute nillable.

**Describe alternatives you've considered**
No alternate is proposed

**LDD Dictionary Version**
For bug fixes, note the LDD dictionary version in question.

**PDS4 IM Version**
1.16

**Need-by Date**
July 20

**Additional context**
Add any other context or examples products

### LINKED ISSUE #254: [ldd-vgr] <fds_scet_epoch_1-4 elements, times incorrectly fail pattern match validation>
**Issue Type**
Using the LABELS section to the right of this window >>>> , please indicate if this is a BUG or ENHANCEMENT request. If unknown, feel free to leave that blank and the LDD Steward can triage appropriately.

**Describe the issue identified (if applicable)**
When validating labels the fds_scet_epoch_1-4 fail the pattern match.  They should not fail, as their patterns do match the required pattern.

  ERROR  [error.label.schema]   line 116, 33: cvc-pattern-valid: Value '1978-08-15T22:21:36.232776Z' is not facet-valid with respect to pattern 'YYYY-MM-DDTHH:MM:SS.SSSSSSZ' for type 'fds_scet_epoch_1'.

**Describe the solution you'd like**
Please update the LDD so that the error does not occur.  

**Describe alternatives you've considered**

**LDD Dictionary Version**

**PDS4 IM Version**
1.16

**Need-by Date**
July 20, 2023

**Additional context**
Add any other context or examples products

### LINKED ISSUE #256: [ldd-vgr] <Update of enumerated values for edit_mode_id, legacy_label_na_exposure_type, legacy_label_wa_exposure_type>
**Issue Type**

**Describe the issue identified (if applicable)**
Please update the enumerated values lists of the following attributes

 edit_mode_id - please add "0:0" to the list  to indicate unknown, as there are instances where this information is not available
 
legacy_label_na_exposure_type - please add "Auto" to the enumeration list for this attribute. "Auto" indicates that the auto-exposure methodology was used to acquire the image.

legacy_label_wa_exposure_type - please add "Auto" to the enumeration list for this attribute. "Auto" indicates that the auto-exposure methodology was used to acquire the image.

**Describe the solution you'd like**
please update the enumeration lists

**Describe alternatives you've considered**
none

**LDD Dictionary Version**
PDS4_VGR_1G00_0100

**PDS4 IM Version**
1.16.

**Need-by Date**
July 22, 2022

**Additional context**
Add any other context or examples products

### PR DESCRIPTION
**Pull request by matthewtiscareno**
v0.5.0.0  2023-07-17

Updated definitions for `spacecraft_clock_start_count` and `spacecraft_clock_stop_count`.

Updated value_data_type and removed pattern for `fds_scet_epoch_1`, `fds_scet_epoch_2`, `fds_scet_epoch_3`, `fds_scet_epoch_4`, and `spacecraft_clock_count_partition`. (resolves pds-data-dictionaries/PDS4-LDD-Issue-Repo#254 and pds-data-dictionaries/PDS4-LDD-Issue-Repo#255)

Updated `nillable_flag` for `earth_received_time`. (resolves pds-data-dictionaries/PDS4-LDD-Issue-Repo#253)

Added enumerated values for `edit_mode_id`, `legacy_label_na_exposure_type`, and `legacy_label_wa_exposure_type`. (resolves pds-data-dictionaries/PDS4-LDD-Issue-Repo#256)

### CODE REVIEW

**Comment by matthewtiscareno** [comment]
@mace-space, @mit3ch, and @cgobat may all be interested in this.

**Comment by matthewtiscareno** [comment]
I'm not sure why the tests failed.  It might be a ASCII problem, but I don't have time to do more on this right at the moment.

File: src/PDS4_VGR_IngestLDD.xml [hunk 1]
```diff
@@ -262,11 +267,10 @@
...
+    <definition>The Voyager spacecraft clock, known as the Flight Data System (FDS), was reset to zero multiple times, dividing each mission into FDS partitions, which count starting at 1.  If there is uncertainty regarding the correct FDS partition (that is, the correct value of spacecraft_clock_count_partition), then this attribute indicates what spacecraft_clock_start_count would be if the partition were 4.<!--Verify that start_date_time is the right thing to say here.--></definition><!--There should be a schematron rule ensuring that at least two of the fds_scet_epoch_N values are used whenever spacecraft_clock_count_partition is zero.-->
     <DD_Value_Domain>
       <enumeration_flag>false</enumeration_flag>
-      <value_data_type>ASCII_Short_String_Collapsed</value_data_type>
-      <pattern>YYYY-MM-DDTHH:MM:SS.SSSSSSZ</pattern>
+      <value_data_type>ASCII_Date_Time_YMD_UTC</value_data_type>
```
Comment (cgobat) [inline src/PDS4_VGR_IngestLDD.xml:267]: Seems like it was indeed an ASCII issue. There were some extra `&ZeroWidthSpace;` characters in the data type value strings, but that's fixed now.

**Review by cgobat — APPROVED** [review]
LGTM!

**Comment by matthewtiscareno** [comment]
Thank you, @cgobat!  I think you've fixed the problems. 

I did clean up the indentation, so that every line has the style you added.  I'll try to keep to that in the future. 

I'd like @katecrombie to review, and then we can merge.

**Review by katecrombie — APPROVED** [review]

**Comment by katecrombie** [comment]
Thanks Matt,

This looks good.  The updated definitions are really helpful.

**Pull request merged**